### ISSUE TITLE
IDs continue incrementing after drop

### ISSUE THREAD

**Issue by stebl** [balderdashy/sails-disk#41]
from stack overflow: http://stackoverflow.com/questions/29926844/sails-waterline-autoincrement

if you set up a model with migrate: 'drop', it will clear the database, but the next created item ID will continue where the old IDs left off.

**Comment by sailsbot** [comment]
Thanks for posting, @stebl.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by ezemoro** [comment]
This is still happening, Any updates about this? @stebl

**Comment by stebl** [comment]
we figured out a work around that ended up being better for us.  we were originally trying to run tests and trying to rely on each suite having known ids 0,1,2,etc when loading things as seed data.  which allows you to test ```PUT /user/3``` for example.  

what we do now is create [barrels fixtures](https://www.npmjs.com/package/barrels) and use the resulting ```barrels.data``` object to get the models and reference them by the index they were seeded in ex. ```barrels.data.user[0]``` returns the POJO of that model.  this works for any ID and ensures the test can always find the info, including if you want to use UUIDs.

it requires a bit of work to get the IDs due to the way barrels works, but it's not too bad.  I'll put the code snippet below to get you and anyone else started.  this is part of a larger function, edited for brevity.

```javascript
seed = {}

async.series([
  function (next) {
    barrels.populate(fixtureName, next);
  },
  function (next) {
    // grab full versions of each model, mostly for their uuids
    async.each(barrels.modelNames, function (modelName, modelNames_callback) {
      var ids = barrels.idMap[modelName];

      if (!ids) {
        return modelNames_callback();
      }

      seed[modelName] = [];

      // iterate each id in series to preserve order
      async.eachSeries(ids, function (id, id_callback) {

        sails.models[modelName]
        .findOne({'id': id})
        .populateAll()
        .then(function (found_model) {
          seed[modelName].push(found_model);
          id_callback();
                    
        })
      }, modelNames_callback);
      
    }, next);
  }], callback);
```
